cgen error for `_ = spawn f()`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.11 5c5566e, press to see full `v doctor` output</summary>

|V full version      |V 0.4.11 13ba08fba28359eed35972f3bc7acfdef42ca79b.5c5566e
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 20.04.6 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
|Memory              |1.75GB/15.05GB
|                    |
|V executable        |/home/delian/v/v
|V last modified time|2025-08-05 13:38:07
|                    |
|V home dir          |OK, value: /home/delian/v
|VMODULES            |OK, value: /home/delian/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/delian/v
|                    |
|env VFLAGS          |"-no-parallel"
|Git version         |git version 2.50.0
|V git status        |weekly.2025.32-2-g5c5566ec-dirty
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|gcc version         |gcc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.31-0ubuntu9.18) 2.31

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  a.v && /home/delian/v/a`
```v
fn f() ! { println('hi') }
_ = spawn f()

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/a.01K1XR37W9V2MYXSBAQETG1N50.tmp.c:1300: error: ',' expected (got "thread")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```

**What did you expect to see?**

a compiled program or a checker error, preferably a compiled program

--- a/vlib/v/markused/walker.v
+++ b/vlib/v/markused/walker.v
@@ -519,6 +519,7 @@ fn (mut w Walker) expr(node_ ast.Expr) {
 			if node.is_expr {
 				w.fn_by_name('free')
 			}
+			w.mark_by_type(w.table.find_or_register_thread(node.call_expr.return_type))
 			w.expr(node.call_expr)
 			w.uses_spawn = true
 

--- a/vlib/v/tests/concurrency/thread_type_test.v
+++ b/vlib/v/tests/concurrency/thread_type_test.v
@@ -1,0 +1,13 @@
+fn f() ! {
+	println('hi')
+}
+
+fn g() int {
+	println('hi')
+	return 23
+}
+
+fn test_main() {
+	_ = go f()
+	_ := spawn g()
+}

